ion-hash/tests/ion_hash_tests.rs

```diff
@@ -119,19 +122,50 @@ fn test_file(file_name: &str) -> IonResult<()> {
 fn test_all<E: Element>(elems: Vec<E>) -> IonResult<()> {
     for case in &elems {
         let case = case.as_struct().expect("test cases are structs");
-        // TODO: support binary ion
-        let ion = case.get("ion").expect("test cases have an `ion` value");
         let expect = case
             .get("expect")
             .expect("test cases have an `expect` value");
-        test_case(ion, expect)?;
+
+        // The case example can be text or binary but not both. Text is already
+        // decoded at this point.
+        //
+        // Implementation note: the arms call into `test_case` directly (vs.
+        // returning an `Element`) because of ownership. In the text case, we
+        // already have a decoded, borrowed Element. In the binary case, we
+        // first need to decode one and thus need to store the decoded Element
+        // somewhere. Thus, the two cases have different types: `&E` vs `E`.
+        // While we could clone the text case, it's just as simple to just
+        // inline the call to `test_case` and avoid the clone.
+        match (case.get("ion"), case.get("10n")) {
+            (Some(text), None) => test_case(text, expect)?,
+            (None, Some(binary)) => {
+                // The sexp contains a binary value without the BVM.
+                let value = seq_to_bytes(binary);
+                // TODO: the binary IVM is pub(crate) in ion-rust
+                let mut bytes = vec![0xE0, 0x01, 0x00, 0xEA];
+                bytes.extend(value);
+
+                let loaded = element_reader().read_all(&bytes)?;
+                let elem = loaded
+                    .into_iter()
+                    .nth(0)
+                    .expect("10n test case should have a single element");
+                test_case(&elem, expect)?;
+            }
+            _ => unreachable!("test cases have one of `ion` or `10n`"),
```

Comment:
Phrasing suggestion to be a bit more explicit:

```suggestion
            _ => unreachable!("test case structs must have either an `ion` field or  an `10n` field"),
```